packages/shared/lib/migration.ts

```diff
@@ -360,16 +360,23 @@ export const findMigrationBundle = (bundleIndex: number): Bundle => {
  */
 export const mineLedgerBundle = (bundleIndex: number, offset: number): Promise<void> =>
     new Promise((resolve, reject) => {
-        const { accounts } = get(wallet)
-
-        const accountId = get(accounts).find((account) => account.index === get(activeProfile).ledgerMigrationCount).id
-
-        api.getMigrationAddress(false, accountId, {
-            onSuccess(response) {
-                resolve(response.payload)
+        api.getAccounts({
+            onSuccess(getAccountsResponse) {
+                api.getMigrationAddress(
+                    false,
+                    getAccountsResponse.payload[get(activeProfile).ledgerMigrationCount].id,
```

Comment:
Adding optional chaining here might prevent a runtime exception (just in case for whatever reason the `activeProfile` is not found), like:
```suggestion
                    getAccountsResponse.payload[get(activeProfile)?.ledgerMigrationCount]?.id,
```

Reply:
What would happen if it is undefined? Error from rust side? We don't want to accidentally generate a migration address on the wrong index

Reply:
I'd imagine just a rust-side error would be handed back, `Invalid type of action received.` or something like that

Reply:
I’ll simulate it